hs_core/models.py

```diff
@@ -3699,16 +3719,16 @@ def replaced_by(self):
         replacedby = self.metadata.relations.all().filter(type='isReplacedBy')
         rlist = []
         for r in replacedby:
-            replacement = r.value
+            citation = r.value
+            res_id = citation[-32:]
             # TODO: This is a mistake. This hardcodes the server on which the URI is created as its URI
-            if replacement.startswith(current_site_url() + "/resource/"):
-                replacement = replacement[-32:]  # strip header
+            res_path = "{}/resource/{}".format(current_site_url(), res_id)
+            if citation.endswith(res_path):
                 try:
-                    rv = get_resource_by_shortkey(replacement, or_404=False)
-                except BaseResource.DoesNotExist:
-                    rv = None
-                if rv is not None:
+                    rv = get_resource_by_shortkey(res_id, or_404=False)
```

Comment:
I think ```_or_404``` should not be set to False since the exception is caught any way and it might get this nonexistent rv added to rlist if the resource does not exist.

Reply:
This is not a change I made. However, I think setting `or_404=False` here is correct since we don't want to get a `Http404` exception but get `BaseResource.DoesNotExist` exception when the resource is missing. This is the exception we are handling in line 3730 so that the rv is not added to the rlist.

Reply:
Agreed after I looked into the implementation for get_resource_by_shortkey() method.